ui/src/dashboards/components/ImportDashboardOverlay.tsx

```diff
@@ -34,39 +51,77 @@ class ImportDashboardOverlay extends PureComponent<Props, State> {
 
     return (
       <Container maxWidth={800}>
-        <Heading title="Import Dashboard" onDismiss={onDismissOverlay} />
-        <Body>
+        <Heading title={this.title} onDismiss={onDismissOverlay} />
+        <Body>{this.renderStep}</Body>
+      </Container>
+    )
+  }
+
+  private get renderStep(): JSX.Element {
+    const {step, importedSources, cells} = this.state
+    const {source, sources} = this.props
+
+    switch (step) {
+      case ImportSteps.IMPORT:
+        return (
           <DragAndDrop
-            submitText="Upload Dashboard"
+            submitText="Continue"
             fileTypesToAccept={this.validFileExtension}
-            handleSubmit={this.handleUploadDashboard}
+            handleSubmit={this.handleContinueImport}
           />
-        </Body>
-      </Container>
-    )
+        )
+      case ImportSteps.MAPPING:
+        return (
+          <ImportDashboardMappings
+            cells={cells}
+            source={source}
+            sources={sources}
+            importedSources={importedSources}
+            onSubmit={this.handleUploadDashboard}
+          />
+        )
+    }
+  }
+
+  private get title(): string {
+    const {step} = this.state
+
+    switch (step) {
+      case ImportSteps.IMPORT:
+        return 'Import Dashboard'
+
+      case ImportSteps.MAPPING:
+        return 'Reconcile Sources'
+    }
   }
 
   private get validFileExtension(): string {
     return '.json'
   }
 
-  private handleUploadDashboard = (
+  private handleContinueImport = (
     uploadContent: string,
     fileName: string
   ): void => {
-    const {notify, onImportDashboard, onDismissOverlay} = this.props
+    const {notify} = this.props
     const fileExtensionRegex = new RegExp(`${this.validFileExtension}$`)
     if (!fileName.match(fileExtensionRegex)) {
       notify(notifyDashboardImportFailed(fileName, 'Please import a JSON file'))
       return
     }
 
     try {
-      const {dashboard} = JSON.parse(uploadContent)
+      const {dashboard, meta} = JSON.parse(uploadContent)
 
       if (!_.isEmpty(dashboard)) {
-        onImportDashboard(dashboard)
-        onDismissOverlay()
+        const cells = _.get(dashboard, 'cells', [])
+        const importedSources = _.get(meta, 'sources', {})
```

Comment:
get deep on these two lines plzz?

Reply:
word. fyi these don't need to be getDeep since they're not nested. has our convention become that we prefer `getDeep` generally?

Reply:
changed - thanks